### PR TITLE
Use enum class instead of plain enum for PrimitiveType

### PR DESCRIPTION
**Pull request by JonnyPtn**
Rebase of #1726

### CODE REVIEW

File: include/SFML/Graphics/PrimitiveType.hpp [hunk 1]
```diff
@@ -36,7 +36,7 @@ namespace sf
 /// and view.
 ///
 ////////////////////////////////////////////////////////////
-enum PrimitiveType
+enum class PrimitiveType
```
Comment (Jim-Marsden) [inline include/SFML/Graphics/PrimitiveType.hpp:39]: I thought there was talk about using enum struct over enum class?

Comment (JonnyPtn) [reply include/SFML/Graphics/PrimitiveType.hpp:39]: As far as I know there's no difference, but `enum class` is the de facto standard

Comment (ChrisThrasher) [reply include/SFML/Graphics/PrimitiveType.hpp:39]: `enum class` is what SFML is already using so let's please keep using that.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by ChrisThrasher — APPROVED** [review]
Glad to see us chipping away at converting more of these old style enumerations into scoped enumerations.

**Review by Bromeon — APPROVED** [review]

**Review by Bambo-Borris — APPROVED** [review]

**Pull request merged**